store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -179,7 +179,11 @@ public CarbonReaderBuilder setEndPoint(String value) {
     if (isTransactionalTable) {
       table = CarbonTable.buildFromTablePath(tableName, "default", tablePath);
     } else {
-      table = CarbonTable.buildDummyTable(tablePath);
+      if (filterExpression != null) {
+        table = CarbonTable.buildTable(tablePath, tableName, true);
+      } else {
```

Comment:
Add a comment about the changes and also can you make a separate method ?
If filterExpression is null, call CarbonTable.buildDummyTable(). else call CarbonTable.buildTable()

because buidTable will not tell whether it is a table with schema or without schema (dummy table).

Reply:
ok, done